internal/domain/usecases/document_usecase.go

```diff
@@ -75,9 +77,41 @@ func (u *documentUsecase) CreateDocument(document *dtos.CreateDocumentDTO) *appe
 }
 
 func (u *documentUsecase) UpdateDocumentByID(ID string, updateMap interface{}) *apperror.AppError {
+	existingDocument, err := u.documentRepository.FindDocumentByID(ID)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			u.logger.Named("UpdateDocumentByID").Error(constant.ErrDocumentNotFound, zap.String("documentID", ID))
+			return apperror.InternalServerError(constant.ErrDocumentNotFound)
+		}
+		u.logger.Named("UpdateDocumentByID").Error(constant.ErrFindDocumentByID, zap.String("documentID", ID), zap.Error(err))
+		return apperror.InternalServerError(constant.ErrFindDocumentByID)
+	}
```

Comment:
I'm not 100% sure about this, but I think `u.documentRepository.UpdateDocumentByID` will return `gorm.ErrRecordNotFound` if there's no documents with that id. So, you don't have to `u.documentRepository.FindDocumentByID` and maybe move `if errors.Is(err, gorm.ErrRecordNotFound)` into `if err := u.documentRepository.UpdateDocumentByID(ID, updateMap)`

Reply:
I'm not sure too bro @bookpanda, please check it kub @D33102

Reply:
I don't think it will return ErrRecordNotFound since when I tested with non-recorded ID, it won't give the error. @bookpanda @wiraphatys

Reply:
![Uploading Screenshot 2567-09-30 at 15.03.48.png…]()

Reply:
Should I do this instead?
<img width="655" alt="Screenshot 2567-09-30 at 15 06 36" src="https://github.com/user-attachments/assets/8fdd8d82-50fc-4cf7-ad31-15f028eb2335">

Reply:
thanks that looks good to me